### PR TITLE
Add ability to login without session #7911

### PR DESCRIPTION
**Pull request by gbbirkisson**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/lib/lib-auth/src/main/resources/lib/xp/auth.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5111596&bid=16807634)

File: modules/lib/lib-auth/src/main/java/com/enonic/xp/lib/auth/LoginHandler.java [hunk 1]
```diff
@@ -29,6 +29,11 @@
 public final class LoginHandler
     implements ScriptBean
 {
+    enum Scope
```
Comment (rymsha) [inline modules/lib/lib-auth/src/main/java/com/enonic/xp/lib/auth/LoginHandler.java]: could it be private?

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/lib/lib-auth/src/main/resources/lib/xp/auth.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5111596&bid=16807634)

File: modules/lib/lib-auth/src/main/java/com/enonic/xp/lib/auth/LoginHandler.java (hunk 1 above)
Comment (gbbirkisson) [reply modules/lib/lib-auth/src/main/java/com/enonic/xp/lib/auth/LoginHandler.java]: Yes, I have changed it.

**Comment by rymsha** [comment]
waiting for @anatol-sialitski to review

**Comment by anatol-sialitski** [comment]
@gbbirkisson could you please make one commit instead of three?

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/lib/lib-auth/src/main/resources/lib/xp/auth.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5111596&bid=16807634)

**Pull request merged**